activation/nipost_test.go

```diff
@@ -106,18 +111,42 @@ func newPoetServiceMock(tb testing.TB) (*mocks.MockPoetProvingServiceClient, *go
 
 type poetDbMock struct {
 	errOn        bool
-	unsubscribed bool
+	subscribed   map[[hash.Size]byte]struct{}
+	unsubscribed map[[hash.Size]byte]struct{}
 }
 
-func (*poetDbMock) SubscribeToProofRef(poetID []byte, roundID string) chan types.PoetProofRef {
+func newPoetDbMock() *poetDbMock {
+	return &poetDbMock{
+		subscribed:   make(map[[hash.Size]byte]struct{}),
+		unsubscribed: make(map[[hash.Size]byte]struct{}),
+	}
+}
+
+func (p *poetDbMock) didSubscribe(poetID []byte) bool {
+	_, ok := p.subscribed[hash.Sum(poetID)]
+	return ok
+}
+
+func (p *poetDbMock) didUnsubscribe(poetID []byte) bool {
+	_, ok := p.unsubscribed[hash.Sum(poetID)]
+	return ok
+}
+
+// A compile time check to ensure that poetDbMock fully implements poetDbAPI.
+var _ poetDbAPI = (*poetDbMock)(nil)
```

Comment:
This is not needed, if the mock doesn't implement the API you will get a compilation error even if this check doesn't exist.